src/test/java/bio/terra/app/controller/RepositoryApiControllerAccessTest.java

```diff
@@ -2,55 +2,56 @@
 
 import static bio.terra.service.configuration.ConfigEnum.SAM_RETRY_INITIAL_WAIT_SECONDS;
 import static bio.terra.service.configuration.ConfigEnum.SAM_TIMEOUT_FAULT;
-import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
 
 import bio.terra.common.category.Integration;
 import bio.terra.integration.DataRepoFixtures;
+import bio.terra.integration.IntegrationTestConfiguration;
 import bio.terra.integration.UsersBase;
 import bio.terra.model.ConfigGroupModel;
 import bio.terra.model.ConfigModel;
 import bio.terra.model.ConfigParameterModel;
-import org.junit.Before;
-import org.junit.Test;
-import org.junit.experimental.categories.Category;
-import org.junit.runner.RunWith;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.http.HttpStatus;
 import org.springframework.test.context.ActiveProfiles;
-import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
 
 /**
  * This is meant to be a very lightweight integration test to make sure that SAM actions are used as
  * expected.
  */
-@RunWith(SpringRunner.class)
-@SpringBootTest
-@AutoConfigureMockMvc
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = IntegrationTestConfiguration.class)
 @ActiveProfiles({"google", "integrationtest"})
-@Category(Integration.class)
-public class RepositoryApiControllerAccessTest extends UsersBase {
+@Tag(Integration.TAG)
+class RepositoryApiControllerAccessTest extends UsersBase {
 
   @Autowired private DataRepoFixtures dataRepoFixtures;
 
-  @Before
+  @BeforeEach
+  @Override
```

Comment:
Why does this need an `@Override`?

Reply:
This `setup` method overrides the `setup` method in `UserBase`. The odd thing here is that this setup method doesn't do anything. The reason why this override is needed is because `UsersBase` itself isn't a test class and doesn't have a `@BeforeEach` annotation on its `setup` method.

This is one of the reasons why I don't like subclassing in test classes. It's mostly used for code sharing (as is done here) and the way it works isn't straightforward. I may PR a change later that moves `UsersBase` to a field instead of a base clsas.